runtime/drivers/admin/admin.go

```diff
@@ -405,6 +407,14 @@ func (h *Handle) checkHandshake(ctx context.Context) error {
 		h.projPath = filepath.Join(h.repoPath, meta.GitSubpath)
 	}
 
+	if meta.ArchiveId != "" {
+		h.ArchiveID = meta.ArchiveId
+	}
+
+	if meta.ArchiveCreatedOn != nil {
+		h.ArchiveCreatedOn = meta.ArchiveCreatedOn.AsTime()
+	}
+
 	if meta.ArchiveDownloadUrl != "" {
 		h.downloadURL = meta.ArchiveDownloadUrl
 		return nil
```

Comment:
Possibility for some cleanup here:

```suggestion
	h.archiveID = meta.ArchiveId
	h.archiveCreatedOn = time.Time{}
	if meta.ArchiveCreatedOn != nil {
		h.archiveCreatedOn = meta.ArchiveCreatedOn.AsTime()
	}
	h.downloadURL = meta.ArchiveDownloadUrl
```